packages/vite/src/node/plugins/css.ts

```diff
@@ -3463,7 +3463,8 @@ export function resolveLibCssFilename(
   }
 
   const packageJson = findNearestPackageData(root, packageCache)?.data
-  const name = packageJson ? getPkgName(packageJson.name) : undefined
+  const name =
+    packageJson && packageJson.name ? getPkgName(packageJson.name) : undefined
```

Comment:
We have `findNearestMainPackageData` function that skips the package.json without `name` field. I think swapping `findNearestPackageData` with it matches the users expectations.
https://github.com/vitejs/vite/blob/e8c783f2d90dec191a8b4daea6b8415a1b345d54/packages/vite/src/node/packages.ts#L161-L162